core/src/main/scala/cats/instances/vector.scala

```diff
@@ -130,6 +130,13 @@ trait VectorInstances extends cats.kernel.instances.VectorInstances {
       final override def traverse[G[_], A, B](fa: Vector[A])(f: A => G[B])(implicit G: Applicative[G]): G[Vector[B]] =
         G.map(Chain.traverseViaChain(fa)(f))(_.toVector)
 
+      final override def updated_[A, B >: A](fa: Vector[A], idx: Long, b: B): Option[Vector[B]] =
+        if (idx >= 0L && idx < fa.size.toLong) {
+          Some(fa.updated(idx.toInt, b))
+        } else {
+          super.updated_(fa, idx, b)
```

Comment:
I think we should just return `None` in the `else` case. In any case `idx < fa.size.toLong` wouldn't work if `size` is bigger than `Int.MaxValue` because then it's overflowing :) but I don't think that's possible anyway.

Reply:
Sorry, just this one last thing and then I think it's good to go!

Reply:
Sorry for the delay, will fix shortly!